google-cloud-spanner/lib/google/cloud/spanner/results.rb

```diff
@@ -85,6 +86,7 @@ def rows
 
           fields = @metadata.row_type.fields
           values = []
+          cached_responses = []
```

Comment:
No, the buffer is not bounded. The requirements document doesn't state how large the buffer should be, or how to measure the buffer size. The node implementation buffers up to 10 responses before responding. I don't see where the python implementation buffers responses. So I left it out for now.